src/System.Management.Automation/help/MUIFileSearcher.cs

```diff
@@ -113,25 +114,42 @@ private void SearchForFiles()
             }
         }
 
-        private string[] GetFiles(string path, string pattern)
-        {
+		private string[] GetFiles(string path, string pattern)
+		{ 
 #if UNIX
             // On Linux, file names are case sensitive, so we need to add
             // extra logic to select the files that match the given pattern.
             ArrayList result = new ArrayList();
             string[] files = Directory.GetFiles(path);
+
+            string regexPattern = pattern.Replace(".", @"\.");
```

Comment:
This actually should not be used. RegEx.Escape will also escape the '*' and '?' which we want them as normal character to replace later.